hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -6637,23 +6655,27 @@ protected RowLock getRowLockInternal(byte[] row, boolean readLock, final RowLock
       }
       rowLockContext.setThreadName(Thread.currentThread().getName());
       success = true;
+      span.setStatus(StatusCode.OK);
       return result;
     } catch (InterruptedException ie) {
       if (LOG.isDebugEnabled()) {
         LOG.debug("Thread interrupted waiting for lock on row: {}, in region {}", rowKey,
           getRegionInfo().getRegionNameAsString());
       }
-      span.addEvent("Interrupted exception getting row lock");
+      TraceUtil.setError(span, ie);
       throw throwOnInterrupt(ie);
     } catch (Error error) {
       // The maximum lock count for read lock is 64K (hardcoded), when this maximum count
       // is reached, it will throw out an Error. This Error needs to be caught so it can
       // go ahead to process the minibatch with lock acquired.
       LOG.warn("Error to get row lock for {}, in region {}, cause: {}", Bytes.toStringBinary(row),
         getRegionInfo().getRegionNameAsString(), error);
+      TraceUtil.setError(span, error);
       IOException ioe = new IOException(error);
-      span.addEvent("Error getting row lock");
       throw ioe;
+    } catch (Throwable t) {
```

Comment:
catch Error then catch Throwable again? A litter wired.

Reply:
This is for recording the span status.

Let me see if we could change to use TraceUtil.trace to increase the readability.